### PR TITLE
[sw/silicon_creator] Harden boot_data_digest_is_valid using shares

### PR DESCRIPTION
**Pull request by alphan**
This PR
* Removes NULL checks from hmac driver functions, and
* Hardens `boot_data_digest_is_valid` using shares.

I will create additional PRs for hardening the rest of `boot_data`.

### CODE REVIEW

**Comment by alphan** [comment]
[disassembly](https://gist.github.com/be0c3112be2cf0072c701b8870fbf723)

File: sw/device/silicon_creator/lib/boot_data.c [hunk 1]
```diff
@@ -67,31 +67,90 @@ static rom_error_t boot_data_digest_compute(const void *boot_data,
                 "`digest` must be the first field of `boot_data_t`.");
 
   hmac_sha256_init();
```
Comment (moidx) [inline sw/device/silicon_creator/lib/boot_data.c:69]: nit: maybe at this point we should have a one shot function as well for `hmac_sha256()`

File: sw/device/silicon_creator/lib/boot_data.c [hunk 2]
```diff
@@ -67,31 +67,90 @@ static rom_error_t boot_data_digest_compute(const void *boot_data,
...
+ * First 7 shares are generated using the `sparse-fsm-encode` script while the
+ * last share is `kHardenedBoolTrue ^ kHardenedBoolFalse ^ kDigestShares[0] ^
+ * ... ^ kDigestShares[7]` so that xor'ing all shares with the initial value of
+ * `is_valid`, i.e. `kHardenedBoolFalse`, produces `kHardenedBoolTrue`.
+ *
+ * Encoding generated with
+ * $ ./util/design/sparse-fsm-encode.py -d 6 -m 7 -n 32 \
+ *     -s 266012770 --language=c
+ *
+ * Hamming distance histogram:
```
Comment (moidx) [inline sw/device/silicon_creator/lib/boot_data.c:86]: I think it is ok to remove the histogram, or just keep the relevant lines.

**Review by moidx — APPROVED** [review]

File: sw/device/silicon_creator/lib/boot_data.c (hunk 1 above)
Comment (alphan) [reply sw/device/silicon_creator/lib/boot_data.c:69]: SG, created #10472.

File: sw/device/silicon_creator/lib/boot_data.c (hunk 2 above)
Comment (alphan) [reply sw/device/silicon_creator/lib/boot_data.c]: Done, removed the histogram, just kept min/max HW/HD.

**Pull request merged**